cli/azd/pkg/project/framework_service.go

```diff
@@ -64,7 +64,7 @@ type FrameworkPackageRequirements struct {
 // restore and build commands
 type FrameworkService interface {
 	// Gets a list of the required external tools for the framework service
-	RequiredExternalTools(ctx context.Context) []tools.ExternalTool
+	RequiredExternalTools(ctx context.Context, serviceConfig *ServiceConfig) []tools.ExternalTool
```

Comment:
> Another alternative would be to make the NewFrameworkService (or whatever method we use to create the implementation instante) to use the serviceConfig during the constructor to set the specific settings for each tool.

Do you know how we'd accomplish this cleanly in the DI world? I thought about it and it didn't seem obvious. The design pattern we've kinda pushed here is that `FrameworkService` is a long-lived daemon, mostly stateless service.

Reply:
I don't know how clean it would be, lol. Probably not at all.
The creation of a framework service depends on creating a service table, which is the array of service config ([]serviceConfig).  We would need to create a registration on DI for it and inject it to the project manager.

So, instead of having the project manager to call:

![image](https://github.com/user-attachments/assets/d6f1ffe7-ecff-4cab-aabd-a51398228d84)

The project manager would just use the service table injected into it.  Then, we can inject the same service table to all the framework service constructors.  The constructor would find the service config and use it.

However, this would prevent us from having 2 services using different configuration. Like a containerapp which we want to build remote and another we want to build locally.  I think that's possible only by Matt's design, using the service config as parameter every time

Reply:
Sort of to @weikanglim's point, it was unclear to me how to do this with the current design - I think there would be a fundamental change what a `FrameworkService` is - and I don't see a super obvious path on how to get from where we are to there.